### ISSUE TITLE
[BUG] sporadic AutoTS failures in CI

### ISSUE THREAD

**Issue by Abhay-Lejith** [sktime/sktime#6795]
**Describe the bug**

Sporadic failures related to AutoTS in CI.
One example:
```
FAILED sktime/forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_update_predict_predicted_index
[AutoTS-0-y:1cols-update_params=True-1-fh=[2 5]] - 
ValueError: Model SeasonalNaive returned NaN for one or more series. fail_on_forecast_nan=True
```
Here is a link to the log:
https://github.com/Abhay-Lejith/sktime/actions/runs/9977939676/job/27609096029#step:3:1846

**Comment by fkiraly** [comment]
> Digging deeper, I am a little confused. It seems to be buried in a _vectorize call? Which doesn't make sense.

Why does this not make sense? When a multivariate forecast is attempted, univariate forecasters - like the `AutoTS` adapter - are called once per variable.

**Comment by winedarksea** [comment]
Guess I didn't realize it was written up as only a Univariate approach here. AutoTS is very much a multivariate modeling framework and people would miss out on the capabilities of the multivariate (VAR, VECM, SectionalMotif, MultivariateMotif, etc) and global style ML or NN models with that approach.

**Comment by yarnabrina** [comment]
Yes, this was noted in my PR review, but I guessed it was merged before closing all threads. The `univariate` behaviour is coming from this tag, which could also be `multivariate` or `both`, and I think for `AutoTS` it needs to vary depending on `model_list` or `model_name`.

https://github.com/sktime/sktime/blob/3a55ca5609fe661cddb147bbcce4ac14ff564b41/sktime/forecasting/autots.py#L210

We should add this support as part of #6773

**Comment by fkiraly** [comment]
> Guess I didn't realize it was written up as only a Univariate approach here. AutoTS is very much a multivariate modeling framework and people would miss out on the capabilities of the multivariate (VAR, VECM, SectionalMotif, MultivariateMotif, etc) and global style ML or NN models with that approach. That could also lead to issues if n_jobs is left as Auto and it tries to parallelize inside a parallelization

@winedarksea, yes, you seem to be correct!
I just looked at the tags, and currently the adapter is:

* univariate only
* ignores exogenous data (there is a "todo" flag added by the author, @MBristle)
* point forecasts only (there is a "todo" flag added by the author, @MBristle))

So I have added these three points as the top todos in the `autots` issue here: https://github.com/sktime/sktime/issues/6773

I also wonder whether the "univariate only" restriction is an accident, since @MBristle has added no "todo" flag to the code.

It could be that if we set the `scitype:y` tag to `"both"`, it might just work out-of-the-box with the current implementation.
We have not released the estimator yet, so we can play around with the code without major repercussions on change and deprecation requirements.

**Comment by fkiraly** [comment]
@winedarksea, I've switched the "univariate only" tag to allow multivariate here: https://github.com/sktime/sktime/pull/6805

let's see what happens, perhaps the original implementation by @MBristle already allows multivariate data to be passed, and this was just disabled by mistake.

**Comment by fkiraly** [comment]
> Yes, this was noted in my PR review, but I guessed it was merged before closing all threads. The `univariate` behaviour is coming from this tag, which could also be `multivariate` or `both`, and I think for `AutoTS` it needs to vary depending on `model_list` or `model_name`.
> 
> https://github.com/sktime/sktime/blob/3a55ca5609fe661cddb147bbcce4ac14ff564b41/sktime/forecasting/autots.py#L210
> 
> We should add this support as part of #6773

@yarnabrina, that could be it - the support being dependent on the model list.

Is there a way to determine whether the model supports multivariate data or not, given the parameters? If yes, the tag can be set dynamically in `__init__`. The higher level API requirement from `sktime` side is that passing multivariate data should never result in an exception, hence we need to know at `__init__` time whether we should pass through or broadcast (because no native multivariate support is present)
